sdk/servicebus/azure-servicebus/migration_guide.md

```diff
@@ -1,164 +1,444 @@
+servicebus_client = ServiceBusClient(fully_qualified_namespace, credential=DefaultAzureCredential())
+```
 
-It should also be noted that many of the helper methods that previously existed on the intermediary client (e.g. QueueClient and `peek()`) now
-exist on the receiver (in the case of `peek_messages()`) or sender itself.  This is to better consolidate functionality and align messaging link lifetime
-semantics with the sender or receiver lifetime.
+### Creating sender and receivers
 
-### Client constructors
+- `QueueClient`, `TopicClient` and `SubscriptionClient` have been replaced with methods to create receivers and senders directly from `ServiceBusClient`.
+- To create a receiver for dead-letter sub queue, `get_deadletter_receiver` has been replaced with `get_<queue/subscritpion>_receiver` which takes an optional parameter `sub_queue` for indication.
 
-| In v0.50 | Equivalent in v7 | Sample |
-|---|---|---|
-| `ServiceBusClient.from_connection_string()`    | `ServiceBusClient.from_connection_string()` | [using credential](https://github.com/Azure/azure-sdk-for-python/tree/master/sdk/servicebus/azure-servicebus/samples/sync_samples/authenticate_client_connstr.py ) |
-| `QueueClient.from_connection_string()`    | `ServiceBusClient.from_connection_string().get_queue_<sender/receiver>()` | [client initialization](https://github.com/Azure/azure-sdk-for-python/tree/master/sdk/servicebus/azure-servicebus/samples/sync_samples/send_queue.py ) |
-| `QueueClient.from_connection_string(idle_timeout=None)` | `ServiceBusClient.from_connection_string.get_queue_receiver(max_wait_time=None)` | [providing a timeout](https://github.com/Azure/azure-sdk-for-python/tree/master/sdk/servicebus/azure-servicebus/samples/sync_samples/session_pool_receive.py) |
+In V0.50:
 
-### Sub-client initialization
+```python
+client = ServiceBusClient.from_connection_string(connstr)
 
-In v0.50, to send or receive messages, a `QueueClient` would be created directly or from the `ServiceBusClient.get_queue` method,
-after which user would call `get_sender` or `get_receiver` methods. Alternatively, `azure.servicebus.send_handler.Sender` and `azure.servicebus.receive_handler.Receiver` were used.
+queue_client = client.get_queue(queue_name)
+queue_sender = queue_client.get_sender()
+queue_receiver = queue_client.get_receiver()
+queue_dead_letter_receiver = queue_client.get_deadletter_receiver()
 
-In v7.0, these are migrated to `ServiceBusSender` and `ServiceBusReceiver`. Note that these are internal sub-clients and should be initialized from the root `ServiceBusClient` as follows.
+topic_client = client.get_topic(topic_name)
+topic_sender = topic_client.get_sender()
 
+subscription_client = client.get_subscription(topic_name, subscription_name)
+subscription_receiver = subscription_client.get_receiver()
+subscription_dead_letter_receiver = subscription_client.get_deadletter_receiver()
+```
+
+In V7:
 ```Python
 with ServiceBusClient.from_connection_string(connstr) as client:
```

Comment:
Any reason to use the `with` construct in the v7 code snippet but not in v0.50?

Reply:
it's because v5 `ServiceBusClient` doesn't support the `with` context manager.
but I tend to keep the usage of `with` for v7 for consistency with other snippets.